ckanext/gdi_userportal/logic/action/translation_utils.py

```diff
@@ -37,32 +49,57 @@ def _get_language() -> str:
     return language
 
 
-def nested_replace(data: Any, replace_dict: Dict) -> Any:
-    """
-    Walks over a json tree, replaces string values starting with 'http' with value-label objects
-    where 'value' is an original field value and `label` a replacement found in replacement_dict or None
-    """
-    if isinstance(data, list):
-        return [nested_replace(item, replace_dict) for item in data]
-
-    if isinstance(data, dict):
-        return {key: nested_replace(value, replace_dict)
-                for key, value in data.items()}
-
-    if isinstance(data, str) and data.startswith("http"):
-        return {"value": data, "label": replace_dict.get(data, None)}
-    else:
-        return data
-
-
-def collect_values_to_translate(data: Any, values_to_translate: List) -> List:
-    """Walks over a json tree and returns a list of string values starting with 'http'"""
-    if isinstance(data, str) and data.startswith("http"):
-        values_to_translate.append(data)
-    elif isinstance(data, list):
-        for item in data:
-            collect_values_to_translate(item, values_to_translate)
-    elif isinstance(data, dict):
-        for key, value in data.items():
-            collect_values_to_translate(value, values_to_translate)
-    return values_to_translate
+def _select_and_append_values(data_item: Dict, fields_list: List, target_list: List) -> List:
+    for key, value in data_item.items():
+        if key in fields_list:
+            if isinstance(value, list):
+                target_list.extend(value)
+            else:
+                target_list.append(value)
+    return target_list
+
+
+def collect_values_to_translate(data: Any) -> List:
```

Comment:
important: you should collect references from `search_facets.*.title` and from `search_facets.*.items[*].name` as well, otherwise there may be facets without translation and the facets' title will never get translated.

Reply:
I made a change to request a a translation for a title each time, not in bulk